modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -46,61 +47,72 @@ class PandasOnRayDataframeVirtualPartition(PandasDataframeAxisPartition):
     instance_type = ray.ObjectRef
     axis = None
 
-    def __init__(self, list_of_blocks, get_ip=False, full_axis=True, call_queue=None):
-        if isinstance(list_of_blocks, PandasOnRayDataframePartition):
-            list_of_blocks = [list_of_blocks]
+    def __init__(
+        self, list_of_partitions, get_ip=False, full_axis=True, call_queue=None
+    ):
+        if isinstance(list_of_partitions, PandasOnRayDataframePartition):
+            list_of_partitions = [list_of_partitions]
         self.full_axis = full_axis
         self.call_queue = call_queue or []
-        # In the simple case, none of the partitions that will compose this
-        # partition are themselves virtual partition. The partitions that will
-        # be combined are just the partitions as given to the constructor.
-        if not any(
-            isinstance(obj, PandasOnRayDataframeVirtualPartition)
-            for obj in list_of_blocks
-        ):
-            self.list_of_partitions_to_combine = list_of_blocks
-            return
-        # Check that all axis are the same in `list_of_blocks`
+        # Check that all virtual partition axes are the same in `list_of_partitions`
         # We should never have mismatching axis in the current implementation. We add this
         # defensive assertion to ensure that undefined behavior does not happen.
         assert (
             len(
                 set(
                     obj.axis
-                    for obj in list_of_blocks
+                    for obj in list_of_partitions
                     if isinstance(obj, PandasOnRayDataframeVirtualPartition)
                 )
             )
-            == 1
+            <= 1
```

Comment:
The same question as for dask execution engine

Reply:
Addressed in the other comment.